sklearn/manifold/tests/test_spectral_embedding.py

```diff
@@ -117,31 +135,46 @@ def test_spectral_embedding_two_components(seed=36):
     true_label[0:n_sample] = 1
 
     se_precomp = SpectralEmbedding(
-        n_components=1, affinity="precomputed", random_state=np.random.RandomState(seed)
+        n_components=1,
+        affinity="precomputed",
+        random_state=np.random.RandomState(seed),
+        eigen_solver=eigen_solver,
     )
-    embedded_coordinate = se_precomp.fit_transform(affinity)
-    # Some numpy versions are touchy with types
-    embedded_coordinate = se_precomp.fit_transform(affinity.astype(np.float32))
-    # thresholding on the first components using 0.
-    label_ = np.array(embedded_coordinate.ravel() < 0, dtype="float")
-    assert normalized_mutual_info_score(true_label, label_) == pytest.approx(1.0)
+    for dtype in [np.float32, np.float64]:
+        embedded_coordinate = se_precomp.fit_transform(affinity.astype(dtype))
+        # thresholding on the first components using 0.
+        label_ = np.array(embedded_coordinate.ravel() < 0, dtype=np.int64)
+        assert normalized_mutual_info_score(true_label, label_) == pytest.approx(1.0)
```

Comment:
Isn't it the goal of the parametrization on `dtype`?
```suggestion
    embedded_coordinate = se_precomp.fit_transform(affinity.astype(dtype))
    # thresholding on the first components using 0.
    label_ = np.array(embedded_coordinate.ravel() < 0, dtype=np.int64)
    assert normalized_mutual_info_score(true_label, label_) == pytest.approx(1.0)
```

Reply:
Whoops. yes indeed.

Reply:
To my recollection, this specific test is not not parametrized, so it's done differently, following an example from test_cluster_qr in
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/cluster/tests/test_spectral.py

Which also BTW has given me the way of treating pyamg
```
try:
    from pyamg import smoothed_aggregation_solver  # noqa

    amg_loaded = True
except ImportError:
    amg_loaded = False
```
Maybe someone want to now edit https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/cluster/tests/test_spectral.py making it consistent with this PR.

Reply:
Anyway, please feel free to edit anything you like, of course. My only interest was to add the float32 support, rather than to beautify the tests.

Reply:
I just have opened https://github.com/scikit-learn/scikit-learn/pull/21564 accordingly.